### PR TITLE
Add doc for basic snapshot repository setup for Azure

### PR DESCRIPTION
**Pull request by charliek17**
Added a Azure how-to example.

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc [hunk 1]
```diff
@@ -439,6 +440,49 @@ Assuming you stored this in a file called `elasticsearch.yaml` you can in both c
...
+      spec:
+        secureSettings:
+         - secretName: snapshot-secret
+
+----
+
+. Now upgrade your cluster to the current configuration.
+
+. From within Kibana you are able to access the Azure Blob Storage under Snapshots. Ensure under `client` for this example to use `default` and `container` as `azurestorageaccount1`.
+
```
[the remark this answers is not in the document]
Comment (charliek17) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:485]: Yeah agree this reads much better. thank you!

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc [hunk 2]
```diff
@@ -439,6 +440,49 @@ Assuming you stored this in a file called `elasticsearch.yaml` you can in both c
 kubectl apply -f elasticsearch.yaml
 ----
 
+[id="{p}-basic-snapshot-azure"]
+=== Basic snapshot repository setup for Azure
+
+. Ensure there is a configured Storage Account in Azure and have created an Access Key for that Storage Account.
```
Comment (naemono) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:446]:
```suggestion
. Ensure there is a Storage Account and an Access Key for that Storage Account configured in Azure.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc [hunk 3]
```diff
@@ -439,6 +440,49 @@ Assuming you stored this in a file called `elasticsearch.yaml` you can in both c
...
++
+. Ensure a Container exists within the Storage Account, in this example we will call it `azuresnaphots`
```
Comment (naemono) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:448]:
```suggestion
. Ensure a Container exists within the Storage Account. For the following example we will call it `azuresnaphots`
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc [hunk 4]
```diff
@@ -439,6 +440,49 @@ Assuming you stored this in a file called `elasticsearch.yaml` you can in both c
...
++
+. Now create a file containing the azure credentials. For this example, name it `azure.client.default.account`. The file name is important as it is reflected in the secure setting. Within this file place your Storage Account name for this example we will use `azurestorageaccount1`.
```
Comment (naemono) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:450]:
```suggestion
. Create a file containing the Azure credentials. For this example, name it `azure.client.default.account`. The file name is important as it is reflected in the secure setting. Within this file place your Storage Account name. For this example we will use `azurestorageaccount1`.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc [hunk 5]
```diff
@@ -439,6 +440,49 @@ Assuming you stored this in a file called `elasticsearch.yaml` you can in both c
...
++
+. Create anouther file containing the azure credentials. For this example, name it `azure.client.default.key`. The file name is important as it is reflected in the secure setting. Within this file place your Storage Account Access Key.
```
Comment (naemono) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:452]:
```suggestion
. Create anouther file containing the Azure credentials. For this example name it `azure.client.default.key`. The file name is important as it is reflected in the secure setting. Within this file place your Storage Account Access Key.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc [hunk 6]
```diff
@@ -439,6 +440,49 @@ Assuming you stored this in a file called `elasticsearch.yaml` you can in both c
...
+
+. Under `podTemplate` place the secret just created into the `- secretName:` field.
+
+[source,yaml,subs="attributes"]
+----
+  nodeSets:
+  - name: elasticsearch
+    count: 3
+    config:
+      node.roles: ["master","data_hot", "data_content", "ingest"]
```
Comment (naemono) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:467]:
```suggestion
      node.roles: ["master", "data_hot", "data_content", "ingest"]
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc [hunk 7]
```diff
@@ -439,6 +440,49 @@ Assuming you stored this in a file called `elasticsearch.yaml` you can in both c
...
+    podTemplate:
+      spec:
+        secureSettings:
+         - secretName: snapshot-secret
+
+----
+
+. Now upgrade your cluster to the current configuration.
+
+. From within Kibana you are able to access the Azure Blob Storage under Snapshots. Ensure under `client` for this example to use `default` and `container` as `azurestorageaccount1`.
```
Comment (naemono) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:484]:
```suggestion
. From within Kibana you should be able to access the Azure Blob Storage under Snapshots. Ensure under `client` to use `default` and under `container` use `azurestorageaccount1`.
```

**Review by naemono — CHANGES_REQUESTED** [review]

File: docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (thbkrkr) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc:485]: I guess based on your comment and the fact that you resolved the conversation, that we can apply the suggestion.

[the remark this answers is not in the document]
Comment (thbkrkr) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc]: @naemono I'm a bit sorry, I made [this suggestion](https://github.com/elastic/cloud-on-k8s/pull/7308#discussion_r1389388644) just before your review. What do you think?

[the remark this answers is not in the document]
Comment (naemono) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc]: Looks great. 👍

**Review by naemono — APPROVED** [review]

**Pull request merged**